### DISCUSSION TITLE
limit_train_batches

### DISCUSSION THREAD

**Discussion by sirtris** [Lightning-AI/pytorch-lightning#13905] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi,
when I use the `limit_train_batches` flag in the trainer and set it to `0.1` does it use the same batches in each epoch or does it randomly sample 10% of the images (in my case) from the training data?

Thx

**Comment by akihironitta** [comment, marked as answer]
It completely depends on your dataloader's arg `shuffle=True|False` (or your sampler specifically).

**Reply by akihironitta** [reply]
> So this means that when I set `shuffle=True` it loads different samples in every epoch, right? (I use the standard torch `DataLoader`, without any fancy sampler.)

@sirtris Yes, it should. Do you have any issue around it?

**Reply by sirtris** [reply]
Ok good to know. Maybe that could be clarified in the documentation.

I was using the argument to experiment with different train set sizes. But in the way it is currently implemented the experiments won't make much sense.  

I think it would be great to have it implemented in a way such that only the same samples get sampled from the train set. Then one could also use the argument for experiments with reduced train set size.
Thanks

**Comment by sirtris** [comment]
So this means that when I set `shuffle=True` it loads different samples in every epoch, right?
(I use the standard torch `DataLoader`, without any fancy sampler.)

**Reply by rohitgr7** [reply]
yes, `shuffle=True` add a random-sampler for you.
